service/models.py

```diff
@@ -15,65 +20,100 @@
+    street_line2 = db.Column(db.String(256))
+    city = db.Column(db.String(64))
+    state = db.Column(db.String(46))
+    country = db.Column(db.String(93))
+    zipcode = db.Column(db.String(20))
+    created_at = db.Column(db.DateTime, nullable=False, default =datetime.utcnow)
+    updated_at = db.Column(db.DateTime, nullable=False, default =datetime.utcnow) 
+
 
     def __repr__(self):
-        return "<YourResourceModel %r id=[%s]>" % (self.name, self.id)
+        return "<Customer %r id=[%s]>" % (self.firstname, self.id)
 
     def create(self):
         """
-        Creates a YourResourceModel to the database
+        Creates a Customer to the database
         """
-        logger.info("Creating %s", self.name)
+        logger.info("Creating %s", self.firstname)
         self.id = None  # id must be none to generate next primary key
         db.session.add(self)
         db.session.commit()
 
     def update(self):
         """
-        Updates a YourResourceModel to the database
+        Updates a Customer to the database
         """
-        logger.info("Saving %s", self.name)
+        logger.info("Saving %s", self.firstname)
+        if not self.id:
+            raise DataValidationError("Update called with empty ID field")
         db.session.commit()
 
     def delete(self):
-        """ Removes a YourResourceModel from the data store """
-        logger.info("Deleting %s", self.name)
+        """ Removes a Customer from the data store """
+        logger.info("Deleting %s", self.firstname)
         db.session.delete(self)
         db.session.commit()
 
     def serialize(self):
-        """ Serializes a YourResourceModel into a dictionary """
-        return {"id": self.id, "name": self.name}
+        """ Serializes a Customer into a dictionary """
+        return {"id": self.id, 
+                "firstname": self.firstname,
+                "lastname": self.lastname,
+                "email": self.email,
+                "phone": self.phone,
+                "street_line1": self.street_line1,
+                "street_line2": self.street_line2,
+                "city": self.city,
+                "state": self.state,
+                "country": self.country,
+                "zipcode": self.zipcode,
+                "created_at":self.created_at,
+                "updated_at":self.updated_at
+                }
 
     def deserialize(self, data):
         """
-        Deserializes a YourResourceModel from a dictionary
+        Deserializes a Customer from a dictionary
 
         Args:
             data (dict): A dictionary containing the resource data
         """
         try:
-            self.name = data["name"]
+            self.firstname = data["firstname"]
+            self.lastname = data["lastname"]
+            self.email = data["email"]
+            self.phone = data["phone"]
+            self.street_line1 = data["street_line1"]
+            self.street_line1 = data["street_line2"]
```

Comment:
self.street_line1 -> self.street_line2. Are you sure your tests are passing with this typo? I think you'll need to add in a test to catch this (and in the case of the other variables not being deserialized too) before fixing it.